debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/MongoDataConverterTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MongoDataConverterTest {
+
+    String record = "{\"address\": {\"building\": \"1007\", \"coord\": [-73.856077, 40.848447], \"street\": \"Morris Park Ave\", \"zipcode\": \"10462\"}, \"borough\": \"Bronx\", \"cuisine\": \"Bakery\", \"grades\": [{\"date\": {\"$date\": 1393804800000}, \"grade\": \"A\", \"score\": 2}, {\"date\": {\"$date\": 1378857600000}, \"grade\": \"A\", \"score\": 6}, {\"date\": {\"$date\": 1358985600000}, \"grade\": \"A\", \"score\": 10}, {\"date\": {\"$date\": 1322006400000}, \"grade\": \"A\", \"score\": 9}, {\"date\": {\"$date\": 1299715200000}, \"grade\": \"B\", \"score\": 14}], \"name\": \"Morris Park Bake Shop\", \"restaurant_id\": \"30075445\"}";
+    BsonDocument val = BsonDocument.parse(record);
+    Set<Entry<String, BsonValue>> entry = val.entrySet();
+    SchemaBuilder builder = SchemaBuilder.struct();
+
+    @Test
+    public void testConvertRecord() {
+
+        for (Entry<String, BsonValue> entry2 : entry) {
+            MongoDataConverter.addFieldSchema(entry2, builder);
+        }
+
+        Schema finalSchema = builder.build();
+        Struct struct = new Struct(finalSchema);
+
+        for (Entry<String, BsonValue> entry5 : entry) {
+            MongoDataConverter.convertRecord(entry5, finalSchema, struct);
+        }
+        Assert.assertEquals(struct.toString(), "Struct{address=Struct{building=1007,coord=[-73.856077, 40.848447],street=Morris Park Ave,zipcode=10462},borough=Bronx,cuisine=Bakery,grades=[Struct{date=1393804800000,grade=A,score=2}, Struct{date=1378857600000,grade=A,score=6}, Struct{date=1358985600000,grade=A,score=10}, Struct{date=1322006400000,grade=A,score=9}, Struct{date=1299715200000,grade=B,score=14}],name=Morris Park Bake Shop,restaurant_id=30075445}");
```

Comment:
Instead of asserting on `toString()`, can we assert on the actual fields? That will make it easier to spot the problematic field in case of assertion failures and also let us define the expected type.